File: BOJ/CPP/단계별로_풀어보기/02_if문/BOJ_2884.cpp
/*
	BOJ 2884 : 알람 시계 

	author : kdgyun
	link https://st-lab.tistory.com
	link https://github.com/kdgyun
	Created by kdgyun
	Copyright © 2021 kdgyun. All rights reserved.
*/

#include <iostream>

using namespace std;

int main(int argc, char const *argv[]) {
	int h; // 시
	int m; // 분

	cin >> h >> m;

	// 만약 입력받은 분이 45분 미만일 경우
	if (m < 45) {
		h--;
		m = 60 - (45 - m);

		// 만약 시(hour)가 음수가 될 경우
		if (h < 0) {
			h = 23;
		}
	} else {
		m = (m - 45);
	}

	cout << h << " " << m;

	return 0;
}

File: BOJ/Java/단계별로_풀어보기/B_02_조건문/BOJ_2884.java
package 단계별로_풀어보기.B_02_조건문;

/**
 * BOJ 2884 : 알람 시계
 * 
 * @author kdgyun
 * {@link https://st-lab.tistory.com}
 * {@link https://github.com/kdgyun}
 *
 */

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;
import java.util.StringTokenizer;

public class BOJ_2884 {

	public static void main(String[] args) throws IOException {

		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

		StringTokenizer st = new StringTokenizer(br.readLine(), " ");
		
		int H = Integer.parseInt(st.nextToken());
		int M = Integer.parseInt(st.nextToken());

		if (M < 45) {
			H--;
			M = 60 - (45 - M);
			if (H < 0) {
				H = 23;
			}
			System.out.println(H + " " + M);
		} else {
			System.out.println(H + " " + (M - 45));
		}
	}
}
